scripts/generatePackageSwift.swift

```diff
@@ -5,33 +5,114 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
-import Foundation
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+import class Foundation.PropertyListDecoder
+import class Foundation.ProcessInfo
+import class Foundation.FileManager
 
+let env = ProcessInfo.processInfo.environment
+
+// This struct is read from the .plist stored at versionDependencies.plist
 struct VersionDeps: Codable {
+    // Versions will always be defined in versionDependencies.plist
+    // Versions are the only reference used when releasing.
     var awsCRTSwiftVersion: String
     var clientRuntimeVersion: String
+    // Branches are not normally defined in versionDependencies.plist,
+    // but may be set during development on a feature branch if desired.
+    // These values override environment vars set on CI.
+    // Branches are ignored when building a release.
+    var awsCRTSwiftBranch: String?
+    var clientRuntimeBranch: String?
+    // Paths may be used to point to paths on a development machine.
+    // They may be set by the developer during testing, but should
+    // never be set outside a development branch.
+    // Use of paths will cause CI to fail.
+    // Paths are ignored when building a release.
+    var awsCRTSwiftPath: String?
+    var clientRuntimePath: String?
+}
+
+let includeProtocolTests: Bool = {
+    env["AWS_SDK_PROTOCOL_CODEGEN_TESTS_BUILT"] != nil
+}()
+
+let releaseInProgress: Bool = {
+    env["AWS_SDK_RELEASE_IN_PROGRESS"] != nil
+}()
+
+struct ProtocolTest {
+    let name: String
+    let path: String
+    let local: Bool
 }
+
+let protocolBasePath = "./codegen/protocol-test-codegen/build/smithyprojections/protocol-test-codegen"
+let protocolBasePathLocal = "./codegen/protocol-test-codegen-local/build/smithyprojections/protocol-test-codegen-local"
```

Comment:
this generated package.swift must not know anything about codegen, here you are in a pure Swift world with no context about the codegen.